Src/PCompiler/CompilerCore/Backend/Java/JavaCompiler.cs

```diff
@@ -31,10 +31,10 @@ public IEnumerable<CompiledFile> GenerateCode(ICompilerConfiguration job, Scope
 
             var generators = new List<JavaSourceGenerator>()
             {
-                new TypesGenerator(Constants.TypesDefnFileName),
-                new EventGenerator(Constants.EventDefnFileName),
-                new MachineGenerator(Constants.MachineDefnFileName),
-                new FFIStubGenerator(Constants.FFIStubFileName)
+                new TypesGenerator(Path.Combine(job.OutputDirectory.FullName, Constants.TypesDefnFileName)),
```

Comment:
Instead of this, can you keep the code as it is and then change the writer to write output to output directory.

Reply:
Resolved in 088a3fad5d48b6b694a515ee95746d274f2e52a1